### PR TITLE
First pass at a mirror system IOC

### PR DESCRIPTION
**Pull request by awalter-bnl**
This pull request relates to my first pass at an IOC server for a 'mirror system', changes to this are expected comments/ suggestions are welcomed. As this is an early pull request, and some on the team are new to git, I think it might be valuable to go through a merge process on this as a learning exercise. 

In addition I am looking for any/all feedback on how this could be improved, and am happy to accept feedback after merging to include in future pull requests.

FYI, once merged the server can be tested by others by forking the repo, then creating a local copy and finally swapping to the directory containing the local copy. Once that is done you can follow the process described in the link below, swapping out the first line
`$ python3 -m caproto.ioc_examples.random_walk --list-pvs`
to 
`$ python3 -m  .\src\ari_sxn_simbeamline\Caproto_servers\Mirror_section.py --list-pvs`

https://caproto.github.io/caproto/master/command-line-client.html

### CODE REVIEW

File: src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py
Comment (Jiemin-Li) [inline src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: Two comments: 

1) For the simulated beamline, will mirror temperature be considered? If so, we probably need to add this PV.

2) I noticed there are only x and y translation movement for the mirror, I assume we also need to provide the z-translation.

Comment (awalter-bnl) [reply src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: @Jiemin-Li , Thanks for the comments. 

> For the simulated beamline, will mirror temperature be considered? If so, we probably need to add this PV.

While we will measure the mirror temperature we may not interact with it at the ophyd/bluesky level. I think we should discuss further if this is needed and leave that for a future PR.

> I noticed there are only x and y translation movement for the mirror, I assume we also need to provide the z-translation.

For the ARI M1 mirror (which I am making this first IOC for) we did not request z-translation. SO as the hardware won't have this axis I think we don't want it in the simulated beamline either.

Let me know if you have further comments/questions

Comment (wen-hu) [reply src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: @Jiemin-Li 
1. Temperature is a good point, but not essential at the beginning.
2. Z translation motion (along the beam) is not required for the M1 mirror.

Comment (wen-hu) [reply src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: @awalter-bnl 
"Ry_coarse" and "Ry_fine" mean yaw motion for the M1 mirror, based on the coordinate system for the ARI mirror. I believe you mean Pitch motion. How about using "Rx_coarse" and "Rx_fine" for Pitch and Fine Pitch motion?

Comment (awalter-bnl) [reply src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: @wen-hu for horizontal bounce mirrors Ry is the pitch. (Y is vertical axis).

Comment (wen-hu) [reply src/ari_sxn_simbeamline/Caproto_servers/Mirror_section.py:1]: @awalter-bnl Thanks, you are correct. Ry is for the pitch in this case.

**Review by wen-hu — APPROVED** [review]

**Review by Jiemin-Li — APPROVED** [review]
All looks good to me!

**Pull request merged**